src/Tes.Runner/Transfer/PartsProducer.cs

```diff
@@ -33,17 +59,34 @@ public PartsProducer(IBlobPipeline blobPipeline, BlobPipelineOptions blobPipelin
     /// <returns></returns>
     public async Task StartPartsProducersAsync(List<BlobOperationInfo> blobOperations, Channel<PipelineBuffer> readBufferChannel)
```

Comment:
Now this method has the obfuscated functionality of skipping the missing files - which is not right from the maintainability perspective. This is another complexity that arises from the handling skipping sources at the library level.